### PR TITLE
Secure dill dump

### PR DESCRIPTION
**Pull request by rizar**
Addresses #154

### CODE REVIEW

**Comment by rizar** [comment]
Update: I added some logging for the case of error, still deferring writing to a different file.

**Comment by rizar** [comment]
Waiting for an LGTM. That does not fully close #154, but significantly improves the situation.

File: blocks/main_loop.py [hunk 1]
```diff
@@ -105,6 +106,10 @@ def run(self):
                 pass
         except TrainingFinish:
             self.log.current_row.training_finished = True
+        except Exception as e:
+            logger.error(traceback.format_exc(e))
+            logger.info("Still trying to finish gracefully")
```
Comment (bartvm) [inline blocks/main_loop.py]: Maybe something slightly more informative, like: "An error occurred during the training. Attempting to run extensions before exiting..."

**Comment by bartvm** [comment]
LGTM (might want to look at the slightly vague log entry)

File: blocks/main_loop.py (hunk 1 above)
Comment (rizar) [inline blocks/main_loop.py]: My idea was that they should have seen the ERROR right above. But okay, I will change that.

Comment (bartvm) [inline blocks/main_loop.py]: Oh, yeah, but it was more about the "running extensions" part. It's not very clear what "finish gracefully" means.

**Pull request merged**